### PR TITLE
Add Teams UI Banners

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR introduces a mechanism to set and manage cookies for dashboard types, enhancing user experience by determining which dashboard version to display based on the cookie value. It includes new components and updates existing layouts to incorporate these features.

### Detailed summary
- Added `DashboardTypeCookieSetter` component to manage dashboard type cookies.
- Integrated `DashboardTypeCookieSetter` in various layouts (`AppShell`, `RootTeamLayout`, `AccountLayout`).
- Created `TeamsUIBanner` and `TryTeamsUIBanner` components for user announcements.
- Updated `getCookie` function to handle errors.
- Refactored `AnnouncementBanner` component for improved styling and functionality.
- Modified `HeroSection` to conditionally render links based on cookie value.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
* **#4936** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4936?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MananTank and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/pages/_app.tsx [hunk 1]
```diff
@@ -260,10 +261,12 @@ const ConsoleApp = memo(function ConsoleApp({
       />
 
       <DashboardRouterTopProgressBar />
+      <DashboardTypeCookieSetter type="old" />
```
Comment (MananTank) [inline apps/dashboard/src/pages/_app.tsx:264]: need to move this elsewhere - can't set "old" here

File: apps/dashboard/src/@/components/DashboardTypeCookieSetter.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+export function DashboardTypeCookieSetter(props: {
+  type: "team" | "old";
+}) {
+  // eslint-disable-next-line no-restricted-syntax
+  useEffect(() => {
+    setCookie("x-dashboard-type", props.type);
+  }, [props.type]);
+
+  return null;
+}
```
Comment (jnsdls) [inline apps/dashboard/src/@/components/DashboardTypeCookieSetter.tsx:15]: can we keep this server-side somehow?

File: apps/dashboard/src/app/components/DashboardTypeBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { AnnouncementBanner } from "../../components/notices/AnnouncementBanner";
+
+export function TeamsUIBanner() {
+  return (
+    <AnnouncementBanner
+      label="Teams UI is still in Beta, Go to Old dashboard for fully functional dashboard"
```
Comment (jnsdls) [inline apps/dashboard/src/app/components/DashboardTypeBanner.tsx:6]: language should be less scary:

"You are exploring the new dashboard. Click here to go back to the legacy dashboard."

File: apps/dashboard/src/app/components/DashboardTypeBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      href="/dashboard"
+      trackingLabel="old-dashboard"
+    />
+  );
+}
+
+export function TryTeamsUIBanner() {
+  return (
+    <AnnouncementBanner
+      label="Explore the new teams UI, Now in Beta"
```
Comment (jnsdls) [inline apps/dashboard/src/app/components/DashboardTypeBanner.tsx:16]: "Explore the new teams dashboard. Now in Beta."

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/@/components/DashboardTypeCookieSetter.tsx (hunk 2 above)
Comment (MananTank) [reply apps/dashboard/src/@/components/DashboardTypeCookieSetter.tsx:15]: the login page does the redirect to /dashboard or /team on client side once the login is done, on the landing page - we change the link to get started based on cookie. so doing this client side is much easier.

File: apps/dashboard/src/app/components/DashboardTypeBanner.tsx (hunk 3 above)
Comment (MananTank) [reply apps/dashboard/src/app/components/DashboardTypeBanner.tsx]: updated

File: apps/dashboard/src/app/components/DashboardTypeBanner.tsx (hunk 4 above)
Comment (MananTank) [reply apps/dashboard/src/app/components/DashboardTypeBanner.tsx]: updated

**Review by jnsdls — APPROVED** [review]

**Pull request merged**